mmdet/core/utils/dist_utils.py

```diff
@@ -67,3 +70,79 @@ def reduce_mean(tensor):
     tensor = tensor.clone()
     dist.all_reduce(tensor.div_(dist.get_world_size()), op=dist.ReduceOp.SUM)
     return tensor
+
+
+def pyobj2tensor(pyobj, device='cuda'):
+    """serialize picklable python object to tensor."""
+    storage = torch.ByteStorage.from_buffer(pickle.dumps(pyobj))
+    return torch.ByteTensor(storage).to(device=device)
+
+
+def tensor2pyobj(tensor):
+    """deserialize tensor to picklable python object."""
+    return pickle.loads(tensor.cpu().numpy().tobytes())
+
+
+@functools.lru_cache()
+def _get_global_gloo_group():
+    """Return a process group based on gloo backend, containing all the ranks
+    The result is cached."""
+    if dist.get_backend() == 'nccl':
+        return dist.new_group(backend='gloo')
+    else:
+        return dist.group.WORLD
+
+
+def all_reduce_dict(py_dict, op='sum', group=None, to_float=True):
+    """Apply all reduce function for python dict object.
+
+    The code is modified from https://github.com/Megvii-
+    BaseDetection/YOLOX/blob/main/yolox/utils/allreduce_norm.py.
+
+    NOTE: make sure that py_dict in different ranks has the same keys and
+    the values should be in the same shape.
+
+    Args:
+        py_dict (dict): dict to be applied all reduce op.
+        op (str): operator, could be 'sum' or 'mean'. Default to 'sum'
+        group (:obj:torch.distributed.group, optional): distributed group,
+            Default to None.
+        to_float (bool): Whether to convert all values of dict to float.
+            Default to True.
+
+    Returns:
+        OrderedDict: reduced python dict object.
+    """
+    _, world_size = get_dist_info()
+    if world_size == 1:
+        return py_dict
+    if group is None:
+        # TODO: Is it necessary?
```

Comment:
Any conclution?

Reply:
Sorry ! No time to experiment.

Reply:
Then change it to `TODO: may try not to use gloo in the future`